### ISSUE TITLE
How to achieve "Detect Anything" functionality?

### ISSUE THREAD

**Issue by jiazhen-code** [IDEA-Research/Rex-Omni#70]
Hi authors,

First of all, thank you for your wonderful and impressive work on Rex-Omni!

I have a question regarding the "Detect Anything" capability, as suggested by the paper's title. When I think of this concept, I'm reminded of models like SAM (Segment Anything Model), which provides an interface to retrieve all instance-level masks, for example, by using multiple keypoints as prompts.

From my understanding, Rex-Omni excels at "referring detection," where a specific text phrase drives the detection of a particular object or a class of objects. However, I am curious about how to use it for a more general, non-referring detection task.

For instance, if I were to use a non-specific prompt such as `Please detect anything in the image.`, would it be possible for Rex-Omni to return all object instances present in the image along with their corresponding coordinates?

To be more specific:
1.  Is there a way to use a non-referring prompt (e.g., "detect all objects") to have Rex-Omni output all instances and their bounding boxes?
2.  If this functionality is not currently implemented, are there any plans to incorporate this kind of general, instance-level detection capability in the future?

Thank you for your time and any clarification you can provide. I'm very excited about the direction of this project.

**Comment by Mountchicken** [comment]
Here is a natural, fluent English translation:

Thank you for your interest in our work.
TL;DR: Our current model does not support this feature, but it should be straightforward to obtain it through fine-tuning.

The Prompt-Free detection approach you mentioned is also known as Universal Detection, where the model automatically identifies and classifies all categories present in an image. There are existing methods that can achieve this, such as DINO-X and Qwen3-VL. 


<img width="2500" height="530" alt="Image" src="https://github.com/user-attachments/assets/b99f31cc-db54-44fb-b2ce-17381237a4c8" />
Demo from DINO-X

<img width="1990" height="1658" alt="Image" src="https://github.com/user-attachments/assets/9fbf9984-d330-467e-a40e-e76f215bf14b" />
Demo from Qwen3-VL

---- 

When training Rex-Omni, we did not include this type of training task, so the model currently does not support this capability.

However, implementing it is quite simple. First, you need to prepare a dataset annotated with detection boxes and categories—this can come from existing detection datasets such as COCO or O365, or data labeled using the [Grounding Data Engine](https://github.com/IDEA-Research/Rex-Omni/tree/master/applications/_2_automatic_grounding_data_engine). The training method for this task is: change the prompt to “Please detect anything in the image.” The ground-truth sequence is then formed by concatenating all categories and bounding boxes.

The main difficulty lies in defining the granularity of the categories. For example, given an image, should the instances be labeled simply as person and dog, or should they be described as a person in a white dress and a running dog? This is an area that requires exploration.

In summary, we have not implemented this feature yet, but it can be achieved by fine-tuning Rex-Omni with SFT.
